tests/visualization_tests/test_edf.py

```diff
@@ -1,106 +1,166 @@
+from io import BytesIO
+from typing import Any
+from typing import Callable
 from typing import List
-from typing import Sequence
+from typing import Union
 
 import numpy as np
 import pytest
 
+import optuna
 from optuna import Study
 from optuna.study import create_study
 from optuna.testing.visualization import prepare_study_with_trials
 from optuna.trial import create_trial
-from optuna.visualization import plot_edf
+from optuna.visualization import plot_edf as plotly_plot_edf
+from optuna.visualization._edf import _get_edf_info
+from optuna.visualization._plotly_imports import _imports as plotly_imports
+from optuna.visualization.matplotlib import plot_edf as plt_plot_edf
+from optuna.visualization.matplotlib._matplotlib_imports import _imports as plt_imports
 
 
-def test_target_is_none_and_study_is_multi_obj() -> None:
+if plotly_imports.is_successful():
+    from optuna.visualization._plotly_imports import go
+
+if plt_imports.is_successful():
+    from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
+
+
+NUM_SAMPLES_X_AXIS = 100
```

Comment:
Might be unnecessary this value and the corresponding test: `len(x_values) == NUM_SAMPLES_X_AXIS`.

Reply:
Thanks. It makes more sense. I move this constant into `visualization/_edf.py`